src/arduino.cc/builder/resolve_library.go

```diff
@@ -55,14 +55,14 @@ func ResolveLibrary(ctx *types.Context, header string) *types.Library {
 		return nil
 	}
 
-	if len(libraries) == 1 {
-		return libraries[0]
-	}
-
 	if markImportedLibraryContainsOneOfCandidates(markImportedLibrary, libraries) {
 		return nil
 	}
 
```

Comment:
Note that this should actually never occur at all, if the header is available, it is never returned as missing by the preprocessor (this was different when include scanning was still regex-based instead of preprocessor-based). Keeping this `if` in might be a good idea, though, but perhaps it should be marked as "just in case" with a comment.